hiero-observability/hiero-metrics-core/docs/metrics_details.md

```diff
@@ -0,0 +1,89 @@
+
+There are two main extensions of the Metric interface:
+- [StatelessMetric](../src/main/java/org/hiero/metrics/api/StatelessMetric.java)<br/>
+Terminal metric that does not provide methods to observe/update data points - only access/read them.
+Updates for such metrics are done via external code. Examples are JVM memory usage metric, CPU usage metric, etc.
+- [StatefulMetric](../src/main/java/org/hiero/metrics/api/core/StatefulMetric.java)<br/>
+This is not a terminal metric (has an abstract builder) which stores data points per unique combination of dynamic label values.
+It knows its data point type and how to instantiate it, so if new dynamic label values are provided,
+it will create a new data point and return it to the client for observation, otherwise return existing data point.
+Stateful metric also knows data point default initializer - an object used to initialize newly created data points (like initial value),
+but API also allows to provide custom initializer when accessing specific data point.
+If no dynamic labels are defined for the metric, then it will have a single data point, which also will be created lazily on first access.
+
+### Metric Types
+
+Supported metric types are defined by enum [MetricType](../src/main/java/org/hiero/metrics/api/core/MetricType.java)
+Here is the table of all metrics available to use:
+
+|                                       Metric                                       |   Type   |                                                            Description                                                             |
+|------------------------------------------------------------------------------------|----------|------------------------------------------------------------------------------------------------------------------------------------|
+| [LongCounter](../src/main/java/org/hiero/metrics/api/LongCounter.java)             | Counter  | Allows to increment `long` (only increasing), should be used also for lower types like `integer`.                                  |
+| [DoubleCounter](../src/main/java/org/hiero/metrics/api/DoubleCounter.java)         | Counter  | Allows to increment `double` (only increasing).                                                                                    |
+| [StatelessMetric](../src/main/java/org/hiero/metrics/api/StatelessMetric.java)     | Gauge    | Requires zero ore more `DoubleSupplier`/`LongSupplier` (with optional labels), which will be called on export.                     |
+| [BooleanGauge](../src/main/java/org/hiero/metrics/api/BooleanGauge.java)           | Gauge    | Allows to set single `boolean` value.                                                                                              |
+| [LongGauge](../src/main/java/org/hiero/metrics/api/LongGauge.java)                 | Gauge    | Gauge to observe and store `long`. May have an accumulator applied to previous and observed values.                                |
```

Comment:
What's core difference between LongCounter and LongGauge, besides the former can only be increased, but the latter can be set to any value? Are they handled differently on export (like the counter is preserved, but the gauge is set to a default value, maybe)?